test/stdlib/PrintFloat.swift.gyb

```diff
@@ -578,6 +584,145 @@ PrintTests.test("Printable_CDouble") {
   expectDescription("-1.0", CDouble(-1.0))
 }
 
+if #available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *) {
+PrintTests.test("Printable_Float16") {
+  func asFloat16(_ f: Float16) -> Float16 { return f }
+
+  // Basic sanity checks:
+  let f = 100.125 as Float16
+  expectEqual("f = 100.1", "f = \(f)")
+
+  expectDescription("0.0", asFloat16(0.0))
+  expectDescription("-0.0", -asFloat16(0.0))
+  expectDescription("0.1", asFloat16(0.1))
+  expectDescription("-0.1", asFloat16(-0.1))
+  expectDescription("1.0", asFloat16(1.0))
+  expectDescription("-1.0", asFloat16(-1.0))
+  expectDescription("1.1", asFloat16(1.1))
+  expectDescription("100.1", asFloat16(100.125))
+  expectDescription("-100.1", asFloat16(-100.125))
+
+  // Standard special numbers:
+  expectDescription("inf", Float16.infinity)
+  expectDescription("-inf", -Float16.infinity)
+  expectDescription("3.14", Float16.pi)
+  expectDescription("65504.0", Float16.greatestFiniteMagnitude)
+#if !arch(arm)
+  expectDescription("6e-08", Float16.leastNonzeroMagnitude)
+#endif
+  expectDescription("6.104e-05", Float16.leastNormalMagnitude)
+
+  // Special cases for the underlying algorithms:
+  // Smallest Float16 that requires 5 digits to print accurately
+  expectDescription("0.00010014", Float16(bitPattern: 0x0690))
+
+  // NaNs require special care in testing:
+  // NaN is printed with additional detail to debugDescription, but not description
+  expectNaN("nan", Float16.nan)
+  expectNaN("nan(0xff)", Float16(nan: 255, signaling: false))
+  expectNaN("nan(0xff)", Float16(bitPattern: 0x7eff))
+  expectNaN("-nan", -Float16.nan)
+  expectNaN("-nan(0xff)", -Float16(nan: 255, signaling: false))
+  expectFailure { expectNaN("snan", Float16.signalingNaN) }
+  expectFailure { expectNaN("-snan", -Float16.signalingNaN) }
+  expectFailure { expectNaN("snan(0xff)", Float16(nan: 255, signaling: true)) }
+  expectFailure { expectNaN("-snan(0xff)", -Float16(nan: 255, signaling: true)) }
+  expectFailure { expectNaN("snan(0xff)", Float16(bitPattern: 0x7dff)) }
+
+  expectEqual("nan", Float16.signalingNaN.description)
+  expectEqual("nan", (-Float16.signalingNaN).description)
+  expectEqual("nan", Float16(nan: 255, signaling: true).description)
+  expectEqual("nan", (-Float16(nan: 255, signaling: true)).description)
+  expectEqual("nan", Float16(bitPattern: 0x7dff).description)
+
+  // Every power of 10 should print with only a single digit '1'
+#if arch(arm)
+  let lowerBound = -7 // Or -4?  Does ARM support subnormals?
+#else
+  let lowerBound = -7
+#endif
```

Comment:
@stephentyrone  -- Should we test subnormals here for ARM?

Reply:
Removed the ARM conditional here.  I also noticed that this test was stopping at 10^2 for not apparent reason; I changed it to go up to 10^4.